projects/com.oracle.truffle.llvm.parser.base/src/com/oracle/truffle/llvm/parser/base/model/LLVMToBitcodeAdapter.java

```diff
@@ -310,4 +313,57 @@ public static ResolvedType unresolveType(VectorType type) {
         ResolvedType elementType = unresolveType(type.getElementType());
         return new ResolvedVectorType(type.getLength(), elementType);
     }
+
+    public static Linkage resolveLinkage(String linkage) {
```

Comment:
I think this method should be in the Linkage enum, perhaps as an overloaded decode(String). Having a toString() that includes those would be nice too. Also, are those really case sensitive?